File: resources/code-template/go/sendD2C.go
package main

import (
	"bytes"
	"fmt"
	"math/rand"
	"net/http"
	"os"
	"time"
)

func main() {
	url := "https://{{iotHubHostName}}/devices/{{deviceId}}/messages/events?api-version=2018-06-30"

	for {
		temperature := 20 + (rand.Float64() * 15)
		humidity := 60 + (rand.Float64() * 20)
		message := fmt.Sprintf(`{"temperature":%f, "humidity":%f}`, temperature, humidity)
		payload := []byte(message)
		req, _ := http.NewRequest("POST", url, bytes.NewBuffer(payload))
		req.Header.Add("authorization", "{{deviceSasToken}}")

		fmt.Printf("Sending message: %s\n", message)
		res, err := http.DefaultClient.Do(req)

		if err != nil {
			fmt.Println(err)
			os.Exit(1)
		} else {
			fmt.Printf("IoT Hub responded to message with status: %d\n", res.StatusCode)
			res.Body.Close()
		}

		time.Sleep(time.Second)
	}
}

File: resources/code-template/ruby/sendD2C.rb
require 'json'
require 'uri'
require 'net/http'
require 'openssl'

$stdout.sync = true
url = URI("https://{{iotHubHostName}}/devices/{{deviceId}}/messages/events?api-version=2018-06-30")
http = Net::HTTP.new(url.host, url.port)
http.use_ssl = true
request = Net::HTTP::Post.new(url)
request["authorization"] = '{{deviceSasToken}}'

while true do
    temperature = 20 + (rand() * 15)
    humidity = 60 + (rand() * 20)
    message = {temperature: temperature, humidity: humidity}.to_json
    request.body = message

    puts "Sending message: " + message
    response = http.request(request)
    puts "IoT Hub responded to message with status: " + response.code
    
    sleep(1)
end
